robot/web/web.go

```diff
@@ -244,16 +247,21 @@ func (svc *webService) StartModule(ctx context.Context) error {
 		googlegrpc.StreamInterceptor(grpc_middleware.ChainStreamServer(streamInterceptors...)),
 		googlegrpc.UnknownServiceHandler(svc.foreignServiceHandler),
 	}
-	svc.modServer = module.NewServer(opts...)
-	if err := svc.modServer.RegisterServiceServer(ctx, &pb.RobotService_ServiceDesc, grpcserver.New(svc.r)); err != nil {
+	server := module.NewServer(opts...)
+	if tcpMode {
+		svc.tcpModServer = server
+	} else {
+		svc.unixModServer = server
+	}
+	if err := server.RegisterServiceServer(ctx, &pb.RobotService_ServiceDesc, grpcserver.New(svc.r)); err != nil {
 		return err
 	}
 
-	if err := svc.initStreamServerForModule(ctx); err != nil {
+	if err := svc.initStreamServerForModule(ctx, server); err != nil {
```

Comment:
I remember being worried about making changes to this section because I wasn't sure how to test these changes; I'll do a reread, then maybe ping a domain expert on streaming